pinot-plugins/pinot-stream-ingestion/pinot-kafka-2.0/src/main/java/org/apache/pinot/plugin/stream/kafka20/KafkaStreamMetadataProvider.java

```diff
@@ -169,14 +172,19 @@ public Map<String, PartitionLagState> getCurrentPartitionLagState(
 
   @Override
   public List<TopicMetadata> getTopics() {
-    Map<String, List<PartitionInfo>> namePartitionsMap = _consumer.listTopics();
-    if (namePartitionsMap == null) {
-      return Collections.emptyList();
+    try (AdminClient adminClient = createAdminClient()) {
```

Comment:
Shall we cache the AdminClient？

Reply:
I was thinking about this and purposely didn't cache it. This API will be called only for 1 internal use case that is also only from the controller. Do you think we should cache it now? 
Resource wise, It does maintain a NetworkClient and can hold on to connections etc.